From 03c892dd77f910220da7621e1596635890f5e146 Mon Sep 17 00:00:00 2001
From: Bruno Haible <bruno@clisp.org>
Date: Sun, 13 Jun 2021 23:26:26 +0200
Subject: [PATCH] maint: Don't require an internet connection for running
 autogen.sh.

* libtool-patches: New directory.
* Makefile.maint (libtool-imported-files): Don't call wget. Take the
patches from libtool-patches/ instead.
---
 ChangeLog                                    |   7 +
 Makefile.maint                               |  17 +-
 libtool-patches/README                       |  12 ++
 libtool-patches/libtool-allow-static.diff    |  65 ++++++
 libtool-patches/libtool-midnightbsd-fix.diff | 293 +++++++++++++++++++++++++++
 libtool-patches/libtool-solaris11-fix.diff   |  31 +++
 6 files changed, 412 insertions(+), 13 deletions(-)
 create mode 100644 libtool-patches/README
 create mode 100644 libtool-patches/libtool-allow-static.diff
 create mode 100644 libtool-patches/libtool-midnightbsd-fix.diff
 create mode 100644 libtool-patches/libtool-solaris11-fix.diff

diff --git a/ChangeLog b/ChangeLog
index eda04ef..1e87b99 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,12 @@
 2021-06-13  Bruno Haible  <bruno@clisp.org>
 
+	maint: Don't require an internet connection for running autogen.sh.
+	* libtool-patches: New directory.
+	* Makefile.maint (libtool-imported-files): Don't call wget. Take the
+	patches from libtool-patches/ instead.
+
+2021-06-13  Bruno Haible  <bruno@clisp.org>
+
 	Prepare for 2.4 release.
 	* VERSION: Set to 2.4.
 
diff --git a/Makefile.maint b/Makefile.maint
index c232432..e8a81f1 100644
--- a/Makefile.maint
+++ b/Makefile.maint
@@ -48,22 +48,13 @@ libtool-imported-files : force
 	  fi; \
 	done
 	rm -rf libtool.tmp
-	: "Allow building statically linked binaries, through LDFLAGS=-static ."
-	wget --execute netrc=off --no-verbose --retry-connrefused --timeout=60 -O libtool-allow-static.diff https://lists.gnu.org/archive/html/bug-libtool/2017-07/txt7Q9LRXv6HS.txt
 	chmod u+w build-aux/ltmain.sh
-	patch build-aux/ltmain.sh < libtool-allow-static.diff
-	rm -f libtool-allow-static.diff
-	: "Fix -export-symbols and -export-symbols-regex support on Solaris 11.3"
-	wget --execute netrc=off --no-verbose --retry-connrefused --timeout=60 -O libtool-solaris11-fix.diff 'https://savannah.gnu.org/patch/download.php?file_id=42205'
+	patch build-aux/ltmain.sh < libtool-patches/libtool-allow-static.diff
 	chmod u+w m4/libtool.m4
-	patch -p1 < libtool-solaris11-fix.diff
-	rm -f libtool-solaris11-fix.diff
-	: "Support creating shared libraries on MidnightBSD"
-	wget --execute netrc=off --no-verbose --retry-connrefused --timeout=60 -O libtool-midnightbsd-fix.diff 'https://savannah.gnu.org/patch/download.php?file_id=50595'
+	patch -p1 < libtool-patches/libtool-solaris11-fix.diff
 	chmod u+w build-aux/ltmain.sh m4/libtool.m4
-	head -n 60 < libtool-midnightbsd-fix.diff | patch -p0 build-aux/ltmain.sh
-	tail -n +227 < libtool-midnightbsd-fix.diff | head -n 56 | patch -p0 m4/libtool.m4
-	rm -f libtool-midnightbsd-fix.diff
+	head -n 60 < libtool-patches/libtool-midnightbsd-fix.diff | patch -p0 build-aux/ltmain.sh
+	tail -n +227 < libtool-patches/libtool-midnightbsd-fix.diff | head -n 56 | patch -p0 m4/libtool.m4
 
 .PHONY : libtool-clean
 libtool-clean : force
diff --git a/libtool-patches/README b/libtool-patches/README
new file mode 100644
index 0000000..0d125d8
--- /dev/null
+++ b/libtool-patches/README
@@ -0,0 +1,12 @@
+This directory contains libtool patches, needed for various platforms.
+
+Commands to fetch them:
+
+: "Allow building statically linked binaries, through LDFLAGS=-static ."
+wget --execute netrc=off --no-verbose --retry-connrefused --timeout=60 -O libtool-allow-static.diff https://lists.gnu.org/archive/html/bug-libtool/2017-07/txt7Q9LRXv6HS.txt
+
+: "Fix -export-symbols and -export-symbols-regex support on Solaris 11.3"
+wget --execute netrc=off --no-verbose --retry-connrefused --timeout=60 -O libtool-solaris11-fix.diff 'https://savannah.gnu.org/patch/download.php?file_id=42205'
+
+: "Support creating shared libraries on MidnightBSD"
+wget --execute netrc=off --no-verbose --retry-connrefused --timeout=60 -O libtool-midnightbsd-fix.diff 'https://savannah.gnu.org/patch/download.php?file_id=50595'
diff --git a/libtool-patches/libtool-allow-static.diff b/libtool-patches/libtool-allow-static.diff
new file mode 100644
index 0000000..14c6685
--- /dev/null
+++ b/libtool-patches/libtool-allow-static.diff
@@ -0,0 +1,65 @@
+diff --git a/build-aux/ltmain.sh b/build-aux/ltmain.sh
+index 0f0a2da..1d0185a 100644
+--- a/build-aux/ltmain.sh
++++ b/build-aux/ltmain.sh
+@@ -3720,6 +3720,7 @@ a program from several object files.
+ The following components of LINK-COMMAND are treated specially:
+ 
+   -all-static       do not do any dynamic linking at all
++  -static           do not do any dynamic linking at all
+   -avoid-version    do not add a version suffix if possible
+   -bindir BINDIR    specify path to binaries directory (for systems where
+                     libraries must be found in the PATH setting at runtime)
+@@ -3746,7 +3747,8 @@ The following components of LINK-COMMAND are treated specially:
+   -R[ ]LIBDIR       add LIBDIR to the runtime path of programs and libraries
+   -shared           only do dynamic linking of libtool libraries
+   -shrext SUFFIX    override the standard shared library file extension
+-  -static           do not do any dynamic linking of uninstalled libtool libraries
++  -static-uninstalled-libs
++                    do not do any dynamic linking of uninstalled libtool libraries
+   -static-libtool-libs
+                     do not do any dynamic linking of libtool libraries
+   -version-info CURRENT[:REVISION[:AGE]]
+@@ -6573,9 +6575,9 @@ func_mode_link ()
+ 	build_old_libs=no
+ 	break
+ 	;;
+-      -all-static | -static | -static-libtool-libs)
++      -all-static | -static | -static-uninstalled-libs | -static-libtool-libs)
+ 	case $arg in
+-	-all-static)
++	-all-static | -static)
+ 	  if test yes = "$build_libtool_libs" && test -z "$link_static_flag"; then
+ 	    func_warning "complete static linking is impossible in this configuration"
+ 	  fi
+@@ -6584,7 +6586,7 @@ func_mode_link ()
+ 	  fi
+ 	  prefer_static_libs=yes
+ 	  ;;
+-	-static)
++	-static-uninstalled-libs)
+ 	  if test -z "$pic_flag" && test -n "$link_static_flag"; then
+ 	    dlopen_self=$dlopen_self_static
+ 	  fi
+@@ -6883,7 +6885,7 @@ func_mode_link ()
+       prevarg=$arg
+ 
+       case $arg in
+-      -all-static)
++      -all-static | -static)
+ 	if test -n "$link_static_flag"; then
+ 	  # See comment for -static flag below, for more details.
+ 	  func_append compile_command " $link_static_flag"
+@@ -7174,9 +7176,9 @@ func_mode_link ()
+ 	continue
+ 	;;
+ 
+-      -static | -static-libtool-libs)
+-	# The effects of -static are defined in a previous loop.
+-	# We used to do the same as -all-static on platforms that
++      -static-uninstalled-libs | -static-libtool-libs)
++	# The effects of -static-uninstalled-libs are defined in a previous
++	# loop.  We used to do the same as -all-static on platforms that
+ 	# didn't have a PIC flag, but the assumption that the effects
+ 	# would be equivalent was wrong.  It would break on at least
+ 	# Digital Unix and AIX.
diff --git a/libtool-patches/libtool-midnightbsd-fix.diff b/libtool-patches/libtool-midnightbsd-fix.diff
new file mode 100644
index 0000000..2b24839
--- /dev/null
+++ b/libtool-patches/libtool-midnightbsd-fix.diff
@@ -0,0 +1,293 @@
+--- build-aux/ltmain.sh.orig	2015-02-15 11:15:12.000000000 -0500
++++ build-aux/ltmain.sh	2018-03-07 23:37:11.000000000 -0500
+@@ -7019,7 +7019,7 @@
+ 	    # These systems don't actually have a C library (as such)
+ 	    test X-lc = "X$arg" && continue
+ 	    ;;
+-	  *-*-openbsd* | *-*-freebsd* | *-*-dragonfly* | *-*-bitrig*)
++	  *-*-openbsd* | *-*-freebsd* | *-*-dragonfly* | *-*-bitrig* | *-*-midnightbsd*)
+ 	    # Do not include libc due to us having libc/libc_r.
+ 	    test X-lc = "X$arg" && continue
+ 	    ;;
+@@ -7039,7 +7039,7 @@
+ 	  esac
+ 	elif test X-lc_r = "X$arg"; then
+ 	 case $host in
+-	 *-*-openbsd* | *-*-freebsd* | *-*-dragonfly* | *-*-bitrig*)
++	 *-*-openbsd* | *-*-freebsd* | *-*-dragonfly* | *-*-bitrig* | *-*-midnightbsd*)
+ 	   # Do not include libc_r directly, use -pthread flag.
+ 	   continue
+ 	   ;;
+@@ -8805,13 +8805,13 @@
+ 	  #
+ 	  case $version_type in
+ 	  # correct linux to gnu/linux during the next big refactor
+-	  darwin|freebsd-elf|linux|osf|windows|none)
++	  darwin|freebsd-elf|midnightbsd-elf|linux|osf|windows|none)
+ 	    func_arith $number_major + $number_minor
+ 	    current=$func_arith_result
+ 	    age=$number_minor
+ 	    revision=$number_revision
+ 	    ;;
+-	  freebsd-aout|qnx|sunos)
++	  freebsd-aout|midnightbsd-aout|qnx|sunos)
+ 	    current=$number_major
+ 	    revision=$number_minor
+ 	    age=0
+@@ -8891,12 +8891,12 @@
+           esac
+ 	  ;;
+ 
+-	freebsd-aout)
++	freebsd-aout | midnightbsd-aout)
+ 	  major=.$current
+ 	  versuffix=.$current.$revision
+ 	  ;;
+ 
+-	freebsd-elf)
++	freebsd-elf | midnightbsd-elf | midnightbsd)
+ 	  func_arith $current - $age
+ 	  major=.$func_arith_result
+ 	  versuffix=$major.$age.$revision
+@@ -9122,7 +9122,7 @@
+ 	  *-*-netbsd*)
+ 	    # Don't link with libc until the a.out ld.so is fixed.
+ 	    ;;
+-	  *-*-openbsd* | *-*-freebsd* | *-*-dragonfly*)
++	  *-*-openbsd* | *-*-freebsd* | *-*-dragonfly* | *-*-midnightbsd*)
+ 	    # Do not include libc due to us having libc/libc_r.
+ 	    ;;
+ 	  *-*-sco3.2v5* | *-*-sco5v6*)
+--- configure.orig	2015-02-15 11:14:34.000000000 -0500
++++ configure	2018-03-07 23:37:48.000000000 -0500
+@@ -5903,7 +5903,7 @@
+     lt_cv_sys_max_cmd_len=8192;
+     ;;
+ 
+-  bitrig* | darwin* | dragonfly* | freebsd* | netbsd* | openbsd*)
++  bitrig* | darwin* | dragonfly* | freebsd* | midnightbsd* | netbsd* | openbsd*)
+     # This has been around since 386BSD, at least.  Likely further.
+     if test -x /sbin/sysctl; then
+       lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
+@@ -6311,7 +6311,7 @@
+   lt_cv_deplibs_check_method=pass_all
+   ;;
+ 
+-freebsd* | dragonfly*)
++freebsd* | dragonfly* | midnightbsd*)
+   if echo __ELF__ | $CC -E - | $GREP __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -10981,7 +10981,7 @@
+       ;;
+ 
+     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
+-    freebsd* | dragonfly*)
++    freebsd* | dragonfly* | midnightbsd*)
+       archive_cmds='$CC -shared $pic_flag -o $lib $libobjs $deplibs $compiler_flags'
+       hardcode_libdir_flag_spec='-R$libdir'
+       hardcode_direct=yes
+@@ -12026,7 +12026,7 @@
+   shlibpath_var=LD_LIBRARY_PATH
+   ;;
+ 
+-freebsd* | dragonfly*)
++freebsd* | dragonfly* | midnightbsd*)
+   # DragonFly does not have aout.  When/if they implement a new
+   # versioning mechanism, adjust this.
+   if test -x /usr/bin/objformat; then
+@@ -13877,7 +13877,7 @@
+     # If you are looking for one http://www.opendarwin.org/projects/dlcompat
+     lt_cv_sys_dlopen_deplibs=yes
+     ;;
+-  freebsd* | dragonfly*)
++  freebsd* | dragonfly* | midnightbsd*)
+     lt_cv_sys_dlopen_deplibs=yes
+     ;;
+   gnu* | linux* | k*bsd*-gnu | kopensolaris*-gnu)
+@@ -15451,7 +15451,7 @@
+         archive_cmds_need_lc_CXX=no
+         ;;
+ 
+-      freebsd* | dragonfly*)
++      freebsd* | dragonfly* | midnightbsd*)
+         # FreeBSD 3 and later use GNU C++ and GNU ld with standard ELF
+         # conventions
+         ld_shlibs_CXX=yes
+@@ -16391,7 +16391,7 @@
+ 	    ;;
+ 	esac
+ 	;;
+-      freebsd* | dragonfly*)
++      freebsd* | dragonfly* | midnightbsd*)
+ 	# FreeBSD uses GNU C++
+ 	;;
+       hpux9* | hpux10* | hpux11*)
+@@ -17315,7 +17315,7 @@
+   shlibpath_var=LD_LIBRARY_PATH
+   ;;
+ 
+-freebsd* | dragonfly*)
++freebsd* | dragonfly* | midnightbsd*)
+   # DragonFly does not have aout.  When/if they implement a new
+   # versioning mechanism, adjust this.
+   if test -x /usr/bin/objformat; then
+@@ -19646,7 +19646,7 @@
+       ;;
+ 
+     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
+-    freebsd* | dragonfly*)
++    freebsd* | dragonfly* | midnightbsd*)
+       archive_cmds_F77='$CC -shared $pic_flag -o $lib $libobjs $deplibs $compiler_flags'
+       hardcode_libdir_flag_spec_F77='-R$libdir'
+       hardcode_direct_F77=yes
+@@ -20481,7 +20481,7 @@
+   shlibpath_var=LD_LIBRARY_PATH
+   ;;
+ 
+-freebsd* | dragonfly*)
++freebsd* | dragonfly* | midnightbsd*)
+   # DragonFly does not have aout.  When/if they implement a new
+   # versioning mechanism, adjust this.
+   if test -x /usr/bin/objformat; then
+@@ -22947,7 +22947,7 @@
+       ;;
+ 
+     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
+-    freebsd* | dragonfly*)
++    freebsd* | dragonfly* | midnightbsd*)
+       archive_cmds_FC='$CC -shared $pic_flag -o $lib $libobjs $deplibs $compiler_flags'
+       hardcode_libdir_flag_spec_FC='-R$libdir'
+       hardcode_direct_FC=yes
+@@ -23782,7 +23782,7 @@
+   shlibpath_var=LD_LIBRARY_PATH
+   ;;
+ 
+-freebsd* | dragonfly*)
++freebsd* | dragonfly* | midnightbsd*)
+   # DragonFly does not have aout.  When/if they implement a new
+   # versioning mechanism, adjust this.
+   if test -x /usr/bin/objformat; then
+--- libltdl/configure.orig	2018-08-17 13:30:38.401401000 -0400
++++ libltdl/configure	2018-08-17 13:34:08.556815000 -0400
+@@ -4788,7 +4788,7 @@
+     lt_cv_sys_max_cmd_len=8192;
+     ;;
+ 
+-  bitrig* | darwin* | dragonfly* | freebsd* | netbsd* | openbsd*)
++  bitrig* | darwin* | dragonfly* | freebsd* | netbsd* | openbsd* | midnightbsd*)
+     # This has been around since 386BSD, at least.  Likely further.
+     if test -x /sbin/sysctl; then
+       lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
+@@ -5196,7 +5196,7 @@
+   lt_cv_deplibs_check_method=pass_all
+   ;;
+ 
+-freebsd* | dragonfly*)
++freebsd* | dragonfly* | midnightbsd*)
+   if echo __ELF__ | $CC -E - | $GREP __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -10005,7 +10005,7 @@
+       ;;
+ 
+     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
+-    freebsd* | dragonfly*)
++    freebsd* | dragonfly* | midnightbsd*)
+       archive_cmds='$CC -shared $pic_flag -o $lib $libobjs $deplibs $compiler_flags'
+       hardcode_libdir_flag_spec='-R$libdir'
+       hardcode_direct=yes
+@@ -11050,7 +11050,7 @@
+   shlibpath_var=LD_LIBRARY_PATH
+   ;;
+ 
+-freebsd* | dragonfly*)
++freebsd* | dragonfly* | midnightbsd*)
+   # DragonFly does not have aout.  When/if they implement a new
+   # versioning mechanism, adjust this.
+   if test -x /usr/bin/objformat; then
+@@ -11069,7 +11069,7 @@
+       need_version=no
+       need_lib_prefix=no
+       ;;
+-    freebsd-*)
++    freebsd-* | midnightbsd*)
+       library_names_spec='$libname$release$shared_ext$versuffix $libname$shared_ext$versuffix'
+       need_version=yes
+       ;;
+@@ -12900,7 +12900,7 @@
+     # If you are looking for one http://www.opendarwin.org/projects/dlcompat
+     lt_cv_sys_dlopen_deplibs=yes
+     ;;
+-  freebsd* | dragonfly*)
++  freebsd* | dragonfly* | midnightbsd*)
+     lt_cv_sys_dlopen_deplibs=yes
+     ;;
+   gnu* | linux* | k*bsd*-gnu | kopensolaris*-gnu)
+--- m4/libtool.m4.orig	2015-01-20 11:15:19.000000000 -0500
++++ m4/libtool.m4	2018-03-07 23:40:12.000000000 -0500
+@@ -1714,7 +1714,7 @@
+     lt_cv_sys_max_cmd_len=8192;
+     ;;
+ 
+-  bitrig* | darwin* | dragonfly* | freebsd* | netbsd* | openbsd*)
++  bitrig* | darwin* | dragonfly* | freebsd* | midnightbsd* | netbsd* | openbsd*)
+     # This has been around since 386BSD, at least.  Likely further.
+     if test -x /sbin/sysctl; then
+       lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
+@@ -2655,7 +2655,7 @@
+   shlibpath_var=LD_LIBRARY_PATH
+   ;;
+ 
+-freebsd* | dragonfly*)
++freebsd* | dragonfly* | midnightbsd*)
+   # DragonFly does not have aout.  When/if they implement a new
+   # versioning mechanism, adjust this.
+   if test -x /usr/bin/objformat; then
+@@ -3488,7 +3488,7 @@
+   lt_cv_deplibs_check_method=pass_all
+   ;;
+ 
+-freebsd* | dragonfly*)
++freebsd* | dragonfly* | midnightbsd*)
+   if echo __ELF__ | $CC -E - | $GREP __ELF__ > /dev/null; then
+     case $host_cpu in
+     i*86 )
+@@ -4317,7 +4317,7 @@
+ 	    ;;
+ 	esac
+ 	;;
+-      freebsd* | dragonfly*)
++      freebsd* | dragonfly* | midnightbsd*)
+ 	# FreeBSD uses GNU C++
+ 	;;
+       hpux9* | hpux10* | hpux11*)
+@@ -5650,7 +5650,7 @@
+       ;;
+ 
+     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
+-    freebsd* | dragonfly*)
++    freebsd* | dragonfly* | midnightbsd*)
+       _LT_TAGVAR(archive_cmds, $1)='$CC -shared $pic_flag -o $lib $libobjs $deplibs $compiler_flags'
+       _LT_TAGVAR(hardcode_libdir_flag_spec, $1)='-R$libdir'
+       _LT_TAGVAR(hardcode_direct, $1)=yes
+@@ -6760,7 +6760,7 @@
+         _LT_TAGVAR(archive_cmds_need_lc, $1)=no
+         ;;
+ 
+-      freebsd* | dragonfly*)
++      freebsd* | dragonfly* | midnightbsd*)
+         # FreeBSD 3 and later use GNU C++ and GNU ld with standard ELF
+         # conventions
+         _LT_TAGVAR(ld_shlibs, $1)=yes
+--- m4/ltdl.m4.orig	2018-08-17 13:23:36.306529000 -0400
++++ m4/ltdl.m4	2018-08-17 13:23:48.981944000 -0400
+@@ -473,7 +473,7 @@
+     # If you are looking for one http://www.opendarwin.org/projects/dlcompat
+     lt_cv_sys_dlopen_deplibs=yes
+     ;;
+-  freebsd* | dragonfly*)
++  freebsd* | dragonfly* | midnightbsd*)
+     lt_cv_sys_dlopen_deplibs=yes
+     ;;
+   gnu* | linux* | k*bsd*-gnu | kopensolaris*-gnu)
diff --git a/libtool-patches/libtool-solaris11-fix.diff b/libtool-patches/libtool-solaris11-fix.diff
new file mode 100644
index 0000000..00559a6
--- /dev/null
+++ b/libtool-patches/libtool-solaris11-fix.diff
@@ -0,0 +1,31 @@
+From 506ce53a03ce4a7f9da9be3dff6af0761f468422 Mon Sep 17 00:00:00 2001
+From: Bruno Haible <bruno@clisp.org>
+Date: Sat, 21 Oct 2017 14:21:15 +0200
+Subject: [PATCH] Fix -export-symbols and -export-symbols-regex support on
+ Solaris 11.3.
+
+On Solaris 11.3 (as opposed to Solaris 11.0), the output of /usr/bin/nm -p
+ob object files contains, for global variables without initializer, the
+letter 'C' (it was 'D' in Solaris 11.0).
+
+m4/libtool.m4 (_LT_CMD_GLOBAL_SYMBOLS): Adjust symcode for Solaris.
+---
+ m4/libtool.m4 | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/m4/libtool.m4 b/m4/libtool.m4
+index 597c604..0fa0c3d 100644
+--- a/m4/libtool.m4
++++ b/m4/libtool.m4
+@@ -3953,7 +3953,7 @@ osf*)
+   symcode='[[BCDEGQRST]]'
+   ;;
+ solaris*)
+-  symcode='[[BDRT]]'
++  symcode='[[BCDRT]]'
+   ;;
+ sco3.2v5*)
+   symcode='[[DT]]'
+-- 
+2.7.4
+
-- 
1.9.1

